tests/unit_tests/blas/include/reference_blas_templates.hpp

```diff
@@ -39,74 +39,2377 @@
+            cblas_zdotc_sub_p(n, x, incx, y, incy, pres);
+    }
+}
+
+static void cblas_cdotu_sub_wrapper(const int n, const void *x, const int incx, const void *y,
+                                    const int incy, void *pres) {
+    if (cblas_library() != NULL) {
+        if (cblas_cdotu_sub_p == NULL)
+            cblas_cdotu_sub_p =
+                (void (*)(const int n, const void *x, const int incx, const void *y, const int incy,
+                          void *pres))GET_FUNC(h_libcblas, "cblas_cdotu_sub");
+        if (cblas_cdotu_sub_p != NULL)
+            cblas_cdotu_sub_p(n, x, incx, y, incy, pres);
+    }
+}
+
+static void cblas_zdotu_sub_wrapper(const int n, const void *x, const int incx, const void *y,
+                                    const int incy, void *pres) {
+    if (cblas_library() != NULL) {
+        if (cblas_zdotu_sub_p == NULL)
+            cblas_zdotu_sub_p =
+                (void (*)(const int n, const void *x, const int incx, const void *y, const int incy,
+                          void *pres))GET_FUNC(h_libcblas, "cblas_zdotu_sub");
+        if (cblas_zdotu_sub_p != NULL)
+            cblas_zdotu_sub_p(n, x, incx, y, incy, pres);
+    }
+}
+
+static int cblas_isamax_wrapper(const int n, const float *x, const int incx) {
+    int isamax_res = 0;
+    if (cblas_library() != NULL) {
+        if (cblas_isamax_p == NULL)
+            cblas_isamax_p = (int (*)(const int n, const float *x, const int incx))GET_FUNC(
+                h_libcblas, "cblas_isamax");
+        if (cblas_isamax_p != NULL)
+            isamax_res = cblas_isamax_p(n, x, incx);
+    }
+    return isamax_res;
+}
+
+static int cblas_idamax_wrapper(const int n, const double *x, const int incx) {
+    int idamax_res = 0;
+    if (cblas_library() != NULL) {
+        if (cblas_idamax_p == NULL)
+            cblas_idamax_p = (int (*)(const int n, const double *x, const int incx))GET_FUNC(
+                h_libcblas, "cblas_idamax");
+        if (cblas_idamax_p != NULL)
+            idamax_res = cblas_idamax_p(n, x, incx);
+    }
+    return idamax_res;
+}
+
+static int cblas_icamax_wrapper(const int n, const void *x, const int incx) {
+    int icamax_res = 0;
+    if (cblas_library() != NULL) {
+        if (cblas_icamax_p == NULL)
+            cblas_icamax_p = (int (*)(const int n, const void *x, const int incx))GET_FUNC(
+                h_libcblas, "cblas_icamax");
+        if (cblas_icamax_p != NULL)
+            icamax_res = cblas_icamax_p(n, x, incx);
+    }
+    return icamax_res;
+}
+
+static int cblas_izamax_wrapper(const int n, const void *x, const int incx) {
+    int izamax_res = 0;
+    if (cblas_library() != NULL) {
+        if (cblas_izamax_p == NULL)
+            cblas_izamax_p = (int (*)(const int n, const void *x, const int incx))GET_FUNC(
+                h_libcblas, "cblas_izamax");
+        if (cblas_izamax_p != NULL)
+            izamax_res = cblas_izamax_p(n, x, incx);
+    }
+    return izamax_res;
+}
```

Comment:
This file is already huge, I suggest putting everything above this point in a separate file (maybe named `reference_blas_wrappers.hpp` or something similar) and including it in `reference_blas_templates.hpp`. I actually wish there was a way to avoid so much wrapping and indirection but probably this is the best we can do.

Reply:
> The problem of using reference blas for both baseline and test persists even after https://github.com/oneapi-src/oneMKL/pull/210.

That's weird. I didn't see the problem in the develop using either gdb or output messages. Could you share what you observed please?

Reply:
This open source product calls oneMKL CPU backend via DPC++ interfaces, which then call cblas symbols. I recall that those symbols were resolving to netlib even after #210. This is what I meant with [this comment](https://github.com/oneapi-src/oneMKL/issues/204#issuecomment-1230806775). But you have investigated this more closely than I did, I might be wrong.

Reply:
I think we might have dodged the conflict. After PR #210, it looks to me that it is the _64 APIs that are called in the mklcpu backend, e.g., `cblas_saxpy_64`.

The wrappers were moved to `reference_blas_wrappers.hpp`. (commit [f17ce6c](https://github.com/oneapi-src/oneMKL/pull/251/commits/f17ce6c48dc27c299a4a3f8dc4842e8d2d5a604b))

Reply:
Oh, that makes sense, the _64 wrappers were a separate change. Thanks!